indigo/docs/guides/howto-hello-indigo-tyrian.md

```diff
@@ -0,0 +1,418 @@
+inject CSS.
+
+Firstly we'll create an `index.html` and `app.js` file that looks like
+[this](https://gist.github.com/hobnob/eaa03bfe2da14562e5f2819078f16d63). You'll
+notice we're using the direct JS output from our build here, which may feel odd.
+What will happen when we run ParcelJS through Yarn is that the HTML will be
+copied to a build directory along with the JS, CSS and any dependant static
+files (such as images) that may be needed.
+
+Now generate an empty `css/main.css`.
+
+Next we'll need to update the `package.json` so that ParcelJS will run when yarn
+starts and copies the static files for Indigo to use. To do this add the
+following to the top of `package.json`:
+
+```json
+"scripts": {
+  "start": "parcel index.html --open --no-cache"
+}
+```
+
+We'll need to add to tell ParcelJS about our static files by adding the
+following to `package.json`:
+
+```json
+"staticFiles": {
+  "staticPath": "assets",
+  "staticOutPath": "assets",
+  "watcherGlob": "**"
+}
+```
+
+We need to add a couple of dependencies. Firstly we'll add the
+dependency for static files by running
+`yarn add parcel-reporter-static-files-copy --dev` from the command line; and
+secondly we'll add `snabbdom` for Tyrian by running `yarn add snabbdom`.
+
+Finally, we need to configure ParcelJS so that it knows how to copy our static
+files. Add a `.parcelrc` file and add the following:
+
+```json
+{
+  "extends": ["@parcel/config-default"],
+  "reporters":  ["...", "parcel-reporter-static-files-copy"]
+}
+```
+
+### Update HelloIndigo.scala
+
+We'll be using the `TyrianSubSystem` as a way of communicating between Indigo
+and Tyrian. Add `import tyrian.TyrianSubSystem` to the imports in
+`HelloIndigo.scala`, and then update the `object` to be a `case class` so that
+it it contains the subsystem as an argument. We can also remove the top level
+export, as we'll no longer need it:
+
+```diff
+- import scala.scalajs.js.annotation.JSExportTopLevel
++ import tyrian.TyrianSubSystem
+
+- @JSExportTopLevel("IndigoGame")
+- object HelloIndigo extends IndigoGame[Unit, Unit, Model, Unit] {
++ final case class HelloIndigo(tyrianSubSystem: TyrianSubSystem[Int]) extends IndigoGame[Unit, Unit, Model, Unit] {
+```
+
+We'll also need to tell Indigo to use the new sub-system, which we can do by
+adding an entry to the `boot` method like so:
+
+```diff
+   def boot(flags: Map[String, String]): Outcome[BootResult[Unit]] =
+     Outcome(
+       BootResult
+         .noData(config)
+         .withAssets(assets)
++        .withSubSystems(tyrianSubSystem)
+     )
+```
+
+### Create a Tyrian App
+
+We'll now develop our initial Tyrian app, which will consist simply of
+a canvas (for Indigo), a counter, and a button, wrapped in a few `div` elements.
+Create a new file called `HelloTyrian.scala` inside the `helloindigo/src` folder
+and add the following contents found
+[here](https://gist.github.com/hobnob/436318b3ae5eed5891ba2b18bb8c264b).
```

Comment:
Again, just put it right here, I'd like to read it in context?

Reply:
And give some commentary.

Reply:
Another advantage, is that we can use Mdoc to check it is correct, you just need to fence the code with `scala mdoc:silent`.